### PR TITLE
Add matrix exercise

### PR DESCRIPTION
**Pull request by BNAndras**
Related to #172.

Any feelings about the string to number conversion? I don't feel like it adds much to the exercise so we could make the expected values be lists of stringified numbers instead. Then the students can focus on manipulating the 2D array.

### CODE REVIEW

File: exercises/practice/matrix/.meta/example.vim [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+function! s:Column(index) dict
+  let l:index = a:index - 1
+  let l:results = []
+  for l:row in self.values
+    call add(l:results, get(l:row, l:index))
+  endfor
+
+  return l:results
+endfunction
```
Comment (kotp) [inline exercises/practice/matrix/.meta/example.vim:29]:
```suggestion
endfunction

```

File: exercises/practice/matrix/matrix.vim [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+"
+" Creates a two-dimensional array that can be queried
+" by a row or a column index, returning the values along
+" that axis.
+"
+function! Matrix(string) abort
+  " your code goes here
+endfunction
```
Comment (kotp) [inline exercises/practice/matrix/matrix.vim:9]:
```suggestion
endfunction

```

**Comment by kotp** [comment]
> Related to #172.
> 
> Any feelings about the string to number conversion? I don't feel like it adds much to the exercise so we could make the expected values be lists of stringified numbers instead. Then the students can focus on manipulating the 2D array.

Yes, there is no reason to convert the string to number.  Hopefully we can make it where there is only the expectation that the elements of a matrix, as represented, can be reported as described.

**Comment by BNAndras** [comment]
Great, I'll update.

**Review by kotp — APPROVED** [review]

**Pull request merged**